examples/01-filter/contouring_multi_plot.py

```diff
@@ -0,0 +1,92 @@
+"""
+.. _contouring_along_axis_example:
+
+Contouring Along Axis
+~~~~~~~~~~~~~~~~~~~~~
+
+Generate contour lines along an axis for the random hills dataset and
+compute data averages within each connected contour line.
+
+This uses :func:`slice_along_axis() <pyvista.DataSet.slice_along_axis>`
+to slice along the Z-Axis of the dataset.
+
+"""
+
+import numpy as np
+import pyvista as pv
+from pyvista import examples
+
+
+###############################################################################
+# Create the plotter and load the dataset
+# 
+pl = pv.Plotter(shape=(2, 2))
+
+mesh_linear = examples.load_random_hills()
+mesh_random = mesh_linear.copy()
+
+###############################################################################
+# Generate the data
+
+# Linear data
+mesh_linear.point_data["DataLinear"] = np.linspace(
+    0, 10, mesh_linear.points.shape[0]
+)
+
+# Random data
+rng = np.random.default_rng(seed=12345)
+mesh_random.point_data["DataRandom"] = rng.integers(
+    low=0, high=10, size=mesh_random.points.shape[0]
+)
+
+###############################################################################
+# Add plots of the raw datasets
+pl.subplot(0, 1)
+mesh_linear.set_active_scalars("DataLinear")
+pl.add_mesh(mesh_linear)
+
+pl.subplot(1, 1)
+mesh_random.set_active_scalars("DataRandom")
+pl.add_mesh(mesh_random)
+
+###############################################################################
+# Average data for each connected component.  For this we first generate
+# contours as separate blocks in a :class:`MultiBlock <pyvista.MultiBlock>`,
+# then for each contour dataset we find connected components and average the
+# scalars within each connected component.  We do all these for both datasets.
+
+n_slices = 10
+
+meshes = mesh_linear, mesh_random
+data_names = "DataLinear", "DataRandom"
+
+for subplot_ind, (mesh, data_name) in enumerate(zip(meshes, data_names)):
+    mesh.set_active_scalars("Elevation")
+    contours = mesh.slice_along_axis(n=n_slices, axis="z")
+    # contours is a MultiBlock with n_slices blocks
+
+    for level_ind, contour in enumerate(contours):
+        connectivity = contour.connectivity()
+        # connectivity is annotated with "RegionId" that identifies components
+        labels = connectivity.point_data["RegionId"]
+        num_labels = np.unique(labels).size
+
+        contour_data_name = data_name + "ContourAverages"
+        connectivity.point_data[contour_data_name] = np.zeros_like(
+            mesh.point_data[data_name], shape=labels.shape
+        )
+
+        for id_ in range(num_labels):
+            data = connectivity[data_name][labels == id_]
+            connectivity[contour_data_name][labels == id_] = np.mean(data)
+
+        pl.subplot(subplot_ind, 0)
```

Comment:
We cannot currently have a `pyvista.Plotter` object used in multiple code blocks.  It is defined in the previous one.

Reply:
Aaah, I would never have guessed that! Thanks, @MatthewFlamm. It makes more sense anyway to group plotting, so I can just refactor that.

Reply:
@MatthewFlamm Again, great catch! Can we add a note to the docs and/or corresponding plotting examples that we cannot use the plotter in multiple code blocks? This would be really helpful to users!

Reply:
@adamgranthendry to be clear what we're talking about is specifically structuring and building the PyVista docs. Users don't really need it, they just see the end product. We _could_ have a "pitfalls for developers" collection of tips somewhere, but I don't think we have a good place for that right now.

Reply:
One thought is that it could go into a subsection of https://docs.pyvista.org/extras/developer_notes.html#contributing.  But it is getting quite long in the tooth already.

Reply:
@adeak @MatthewFlamm My bad...I thought this was about how you can't run `p.plot()` multiple times in the same program.

Reply:
@adamgranthendry side note: you can do `pl.show(auto_close=False)` which will keep the plotter alive after closing, allowing you to run `pl.show()` again.

Reply:
@adeak Thanks for the tip!